javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,130 +1,149 @@
-      let nameLI = document.createElement("LI");
-      nameLI.innerHTML = `Procuct: ${product.name}`;
-      productUL.append(nameLI);
-      let priceLI = document.createElement("LI");
-      priceLI.innerHTML = `Price: ${product.price}`;
-      productUL.append(priceLI);
-    });
+    getUser = async (userID) => {
+      try {
+        const baseUrl = "https://jsonplaceholder.typicode.com/users/";
+        const query = userID;
+        return await getData(baseUrl, query);
+      } catch (error) {
+        console.log(error);
+      }
+    };
   }
 
-  getUser = async () => {
+  getData = async (baseUrl, query) => {
     try {
-      let url = `https://jsonplaceholder.typicode.com/users/1`;
-      const response = await fetch(url);
+      const response = await fetch(`${baseUrl}${query}`);
       if (response.status >= 200 && response.status < 300) {
-        let user = response;
-        user = user.json();
-        return user;
+        const data = response.json();
+        return data;
       } else {
         throw new Error(`HTTP error! status: ${response.status}`);
       }
     } catch (error) {
       console.log(error);
     }
   };
-}
-
-const shoppingCart = new ShoppingCart();
 
-const flatscreen = new Product("flat-screen", 1000);
-const tablet = new Product("tablet", 2000);
-const notebook = new Product("notebook", 3000);
-const mobilephone = new Product("mobilephone", 4000);
-const pc = new Product("pc", 5000);
-
-shoppingCart.addProduct(flatscreen);
-shoppingCart.addProduct(tablet);
-shoppingCart.addProduct(notebook);
-shoppingCart.addProduct(mobilephone);
-shoppingCart.addProduct(pc);
-
-shoppingCart.removeProduct(flatscreen);
-
-console.log(shoppingCart.getTotal());
-console.log(shoppingCart.searchProduct("tablet"));
+  async function getCurrencyRate(amountToConvert, convertToCurrency) {
+    const baseUrl = `https://v6.exchangerate-api.com/v6/84a875e2ff2c28d158651393/latest/`;
+    const query = defaultCurrency;
+    const data = await getData(baseUrl, query);
+    const currenciesArray = data.conversion_rates;
+    const convertedPrice = (amountToConvert * currenciesArray[`${convertToCurrency}`]).toFixed(2);
+    console.log(`${convertedPrice}  ${convertToCurrency}`);
+    return `${convertedPrice}  ${convertToCurrency}`;
+  }
 
-async function displayEverything() {
-  let user = await shoppingCart.getUser();
-  const userName = document.createElement("H2");
-  userName.textContent = user.name;
-  document.body.append(userName);
-  shoppingCart.renderProducts();
-  displayTotalPrice();
-}
+  async function displayEverything() {
+    await displayUserInfo();
+    shoppingCart.renderProducts();
+    displayTotalPrice();
+  }
 
-function displayTotalPrice() {
-  const totalPrice = document.createElement("h2");
-  totalPrice.textContent = `Total: ${shoppingCart.getTotal()}`;
-  document.body.appendChild(totalPrice);
-}
+  async function displayUserInfo() {
+    let user = await shoppingCart.getUser(1);
```

Comment:
user '1' is hardcoded. Should be more flexible.

Reply:
async getUser(userID){
}
getUser expects an ID. I need to specify somewhere which user I want.
No clue how to make it more flexible.

Reply:
Sorry indeed it's flexible enough in the function declaration, you only call it as "getUser(1)", which is OK. I think the homework assignment itself isn't clear on what to do with that.